hub/core/dataset/dataset.py

```diff
@@ -387,6 +387,9 @@ def _load_version_info(self):
 
     def _lock(self, err=False):
         storage = get_base_storage(self.storage)
+
+        # temporarily disable read only on base storage, to try to acquire lock, if exception, it will be again made readonly
+        storage.disable_readonly()
```

Comment:
if read_only is set to False, then on line 396, not self.read_only will be True and that check is not necessary?

Also, if the process quits after L392, then is read_only set to False?

How about removing the check for self.read_only, and moving this statement into the try block so there's a better chance of read_only set back to True?

Reply:
self.read_only and storage.read_only are different. Here we're only changing read_only for underlying storage, self.read_only will still be in its original state.

although yes, moving it within the try seems better

Reply:
for my understanding, where is storage.read_only set to True again?

Reply:
if an exception is raised, self.read_only will be set to True in the except block, which will also set it for underlying storage